crates/sui-framework/src/natives/crypto.rs

```diff
@@ -99,3 +108,191 @@ pub fn bls12381_verify_g1_sig(
         Err(_) => Ok(NativeResult::ok(cost, smallvec![Value::bool(false)])),
     }
 }
+
+/// Native implemention of Bulletproofs range proof in public Move API, see crypto.move for specifications.
+pub fn verify_range_proof(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 2);
+
+    let bit_length = pop_arg!(args, u64);
+    let commitment_bytes = pop_arg!(args, Vec<u8>);
+    let proof_bytes = pop_arg!(args, Vec<u8>);
+    let cost = native_gas(context.cost_table(), NativeCostIndex::EMPTY, 0);
+
+    let proof = if let Ok(val) = BulletproofsRangeProof::from_bytes(&proof_bytes[..]) {
+        val
+    } else {
+        return Ok(NativeResult::err(cost, INVALID_BULLETPROOF));
+    };
+
+    let commitment = if let Ok(val) = PedersenCommitment::from_bytes(&commitment_bytes[..]) {
+        val
+    } else {
+        return Ok(NativeResult::err(cost, INVALID_RISTRETTO_GROUP_ELEMENT));
+    };
+
+    match proof.verify_bit_length(&commitment, bit_length as usize, b"sui") {
```

Comment:
I'd put `b"sui" as const if possible (not to have it hidden inside the function.